recipes/protobuf/all/patches/upstream-pr-4620-rpath-management.patch

```diff
@@ -0,0 +1,49 @@
+diff --git a/cmake/CMakeLists.txt b/cmake/CMakeLists.txt
+index f159f60..8e945ff 100644
+--- a/cmake/CMakeLists.txt
++++ b/cmake/CMakeLists.txt
+@@ -8,6 +8,12 @@ endif()
+ # CMake policies
+ cmake_policy(SET CMP0022 NEW)
+ 
++# On MacOS use @rpath/ for target's install name prefix path
++if (POLICY CMP0042)
++  cmake_policy(SET CMP0042 NEW)
++endif ()
++# Clear VERSION variables when no VERSION is given to project()
++
+ # Project
+ project(protobuf C CXX)
+ 
+diff --git a/cmake/install.cmake b/cmake/install.cmake
+index 441bf55..ee17601 100644
+--- a/cmake/install.cmake
++++ b/cmake/install.cmake
+@@ -13,6 +13,13 @@ foreach(_library
+     PROPERTY INTERFACE_INCLUDE_DIRECTORIES
+     $<BUILD_INTERFACE:${protobuf_source_dir}/src>
+     $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>)
++  if (UNIX AND NOT APPLE)
++    set_property(TARGET ${_library}
++      PROPERTY INSTALL_RPATH "$ORIGIN")
++  elseif (APPLE)
++    set_property(TARGET ${_library}
++      PROPERTY INSTALL_RPATH "@loader_path")
++  endif()
+   install(TARGETS ${_library} EXPORT protobuf-targets
+     RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} COMPONENT ${_library}
+     LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT ${_library}
+@@ -21,6 +28,13 @@ endforeach()
+ 
+ install(TARGETS protoc EXPORT protobuf-targets
+   RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} COMPONENT protoc)
++if (UNIX AND NOT APPLE)
++  set_property(TARGET protoc
++    PROPERTY INSTALL_RPATH "$ORIGIN/../lib")
++elseif (APPLE)
++  set_property(TARGET protoc
++    PROPERTY INSTALL_RPATH "@loader_path/../lib")
++endif()
+ 
+ install(FILES ${CMAKE_CURRENT_BINARY_DIR}/protobuf.pc ${CMAKE_CURRENT_BINARY_DIR}/protobuf-lite.pc DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig")
+ 
```

Comment:
It can avoided, test_v1_package should be fixed instead:

```python
from conans import ConanFile, CMake, tools
import os


class TestPackageConan(ConanFile):
    settings = "os", "arch", "compiler", "build_type"
    generators = "cmake", "cmake_find_package_multi"
    test_type = "explicit"

    def requirements(self):
        self.requires(self.tested_reference_str)

    def build_requirements(self):
        if hasattr(self, "settings_build") and cross_building(self):
            self.build_requires(self.tested_reference_str)

    def build(self):
        with tools.no_op() if hasattr(self, "settings_build") and cross_building(self) else tools.run_environment(self):
            cmake = CMake(self)
            cmake.definitions["protobuf_LITE"] = self.options["protobuf"].lite
            cmake.configure()
            cmake.build()

    def test(self):
        if not tools.cross_building(self):
            self.run("protoc --version", run_environment=True)
            self.run(os.path.join("bin", "test_package"), run_environment=True)
```